app/controllers/agent/ws.go

```diff
@@ -126,14 +125,18 @@ func (c Agent) Data(conn *websocket.Conn) revel.Result {
 		return c.Error(err, "Agent.Data: dbm.Open")
 	}
 	defer dbm.Close()
-	dbh := qan.NewMySQLMetricWriter(dbm, shared.QueryAbstracter, &dbStats)
+
+	// create instance handler
+	ih := instance.NewMySQLHandler(dbm)
```

Comment:
I need to inject instance handler to get `subsystem`.

There was actually already call to db to get instance id so I replaced it with call that fetches not only instance id but whole instance so I get also instance.subsytem for free.